src/PackageManagement.VisualStudio/Utility/EnvDTEProjectUtility.cs

```diff
@@ -1257,7 +1257,10 @@ public static bool SupportsINuGetProjectSystem(EnvDTEProject envDTEProject)
         /// </summary>
         public static async Task<bool> HasBuildIntegratedConfig(EnvDTEProject project)
         {
-            return await ContainsFile(project, BuildIntegratedProjectUtility.ProjectConfigFileName);
+            var projectNameConfig = BuildIntegratedProjectUtility.GetProjectConfigWithProjectName(project.Name);
+
+            return await ContainsFile(project, BuildIntegratedProjectUtility.ProjectConfigFileName)
```

Comment:
can't this calls throw, can you please await the call and assign to local variable for each and then do the return, also make sure to use try catch?